src/checks/#cc4a#modern_language.clas.abap

```diff
@@ -0,0 +1,1157 @@
+                statement_index TYPE i
+      RETURNING VALUE(findings) TYPE if_ci_atc_check=>ty_findings.
+    METHODS analyze_create_object
+      IMPORTING procedure       TYPE if_ci_atc_source_code_provider=>ty_procedure
+                statement_index TYPE i
+      RETURNING VALUE(findings) TYPE if_ci_atc_check=>ty_findings.
+    METHODS analyze_call_method
+      IMPORTING procedure       TYPE if_ci_atc_source_code_provider=>ty_procedure
+                statement_index TYPE i
+      RETURNING VALUE(findings) TYPE if_ci_atc_check=>ty_findings.
+    METHODS analyze_exporting_receiving
+      IMPORTING procedure       TYPE if_ci_atc_source_code_provider=>ty_procedure
+                statement_index TYPE i
+      RETURNING VALUE(findings) TYPE if_ci_atc_check=>ty_findings.
+    METHODS analyze_text_assembly
+      IMPORTING procedure       TYPE if_ci_atc_source_code_provider=>ty_procedure
+                statement_index TYPE i
+      RETURNING VALUE(findings) TYPE if_ci_atc_check=>ty_findings.
+    METHODS check_quickfixes
+      IMPORTING quickfixes      TYPE REF TO cl_ci_atc_quickfixes
+                procedure       TYPE if_ci_atc_source_code_provider=>ty_procedure
+                statement_index TYPE i
+      CHANGING  findings        TYPE if_ci_atc_check=>ty_findings.
+    METHODS add_finding
+      IMPORTING quickfixes      TYPE REF TO cl_ci_atc_quickfixes OPTIONAL
+                procedure       TYPE if_ci_atc_source_code_provider=>ty_procedure
+                statement_index TYPE i
+                code            TYPE cl_ci_atc_quickfixes=>ty_quickfix_code
+                pseudo_comment  TYPE string
+      CHANGING  findings        TYPE if_ci_atc_check=>ty_findings.
+    METHODS is_used
+      IMPORTING
+        procedure     TYPE if_ci_atc_source_code_provider=>ty_procedure
+        from_index    TYPE i
+        full_name     TYPE string
+      RETURNING
+        VALUE(result) TYPE abap_bool.
+
+    METHODS check_remove_exporting
+      IMPORTING
+        statement     TYPE if_ci_atc_source_code_provider=>ty_statement
+        token_idx     TYPE i
+      RETURNING
+        VALUE(result) TYPE abap_bool.
+    METHODS get_value
+      IMPORTING token         TYPE if_ci_atc_source_code_provider=>ty_token
+      RETURNING VALUE(result) TYPE string
+      RAISING   lcx_error.
+    METHODS append_token
+      IMPORTING token  TYPE if_ci_atc_source_code_provider=>ty_token
+      CHANGING  result TYPE string.
+    METHODS append_tokens
+      IMPORTING tokens          TYPE if_ci_atc_source_code_provider=>ty_tokens
+                VALUE(from_idx) TYPE i OPTIONAL
+                VALUE(to_idx)   TYPE i OPTIONAL
+      CHANGING  result          TYPE string.
+    METHODS get_receiving_infos
+      IMPORTING
+        tokens        TYPE if_ci_atc_source_code_provider=>ty_tokens
+      EXPORTING
+        receiving_idx TYPE i
+        end_idx       TYPE i
+        result_line   TYPE string.
```

Comment:
We should adhere to our own [guidelines](https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#prefer-returning-to-exporting): Use RETURNING where possible, and here's it's easily possible - just return a structure.